### PR TITLE
Dependencies Checker

### PR DESCRIPTION
**Pull request by honzajavorek**
Not really proud on the code, but it works as a prototype. I didn't anticipate it's going to be that non-trivial, but still... no rocket science.

The most fragile part is the parsing of the copyright notice from the license text, but so far it worked for all packages with correct license I tried.

The docstring contains also things this tool doesn't do yet. Those are marked as `TODO`. They're ideas on what could/should be added next.

Also:

- Added Circle CI config
- Removed deprecation warnings from tests (assert_equals => assert_equal)

### CODE REVIEW

File: blackbelt/dependencies.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    'licenseText', 'licenseModified', 'licenseFile', 'path',
+]
+
+# https://github.com/davglass/license-checker#custom-format
+LICENSE_CHECKER_FORMAT = {}
+for key in LICENSE_CHECKER_FORMAT_KEYS:
+    LICENSE_CHECKER_FORMAT[key] = None
+
+# https://confluence.oraclecorp.com/confluence/display/CORPARCH/Licenses+Eligible+for+Pre-Approval+-+Internal+and+Hosted+Use
+# https://confluence.oraclecorp.com/confluence/display/CORPARCH/Licenses+Eligible+for+Pre-Approval+-+Distribution
```
Comment (Almad) [inline blackbelt/dependencies.py]: :notsureif: confluence links in OSS software...that said, blackbelt is probably kinda falling off the general usage

Comment (honzajavorek) [reply blackbelt/dependencies.py]: Hmm. Yeah... maybe it was never really something generally used :) Do we want to close-source it? The license list is specific to Oracle, not configurable.

File: blackbelt/dependencies.py [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+
+# https://github.com/davglass/license-checker#custom-format
+LICENSE_CHECKER_FORMAT_KEYS = [
+    'name', 'repository', 'dependencyPath', 'url', 'version', 'licenses',
+    'licenseText', 'licenseModified', 'licenseFile', 'path',
+]
+
+# https://github.com/davglass/license-checker#custom-format
+LICENSE_CHECKER_FORMAT = {}
```
Comment (char0n) [inline blackbelt/dependencies.py]: Can be simplified using dict comprehension
```python
LICENSE_CHECKER_FORMAT = {i: None for i in LICENSE_CHECKER_FORMAT_KEYS}
```
Just uncovering my older python knowhow so pls bear with me ;]

File: blackbelt/dependencies.py [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+        click.secho(heading, bold=True, fg=color)
+        missing = False
+
+        for details in problematic_licenses:
+            reasons = ', '.join(details['not_pre_approved_reasons'])
+            missing = missing or 'missing' in reasons
+
+            line = click.style('{name} {version} ({licenses})'.format(**details), bold=True)
+            line += ' - ' + reasons
+            click.echo(line)
```
Comment (char0n) [inline blackbelt/dependencies.py]:
```python
click.echo('{0} - {1}'.format(line, reasons);

File: blackbelt/dependencies.py [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+        heading = '\nProblematic Licenses: {0}'.format(len(problematic_licenses))
+        click.secho(heading, bold=True, fg=color)
+        missing = False
+
+        for details in problematic_licenses:
+            reasons = ', '.join(details['not_pre_approved_reasons'])
+            missing = missing or 'missing' in reasons
+
+            line = click.style('{name} {version} ({licenses})'.format(**details), bold=True)
+            line += ' - ' + reasons
```
Comment (char0n) [inline blackbelt/dependencies.py]: would drop this one and replace it with the next one

File: blackbelt/dependencies.py [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+#   License field?
+# - If there are multiple dependencies licensed under the same terms,
+#   do I need to repeat those terms for every dependency?
+def create_licenses_list(top_level_details, fourth_party_licenses):
+    sections = []
+
+    sections.append(top_level_details['copyright_notice'])
+    sections.append(top_level_details['license_text'])
+
+    def key_fn(details):
```
Comment (char0n) [inline blackbelt/dependencies.py]: Cannot be replaced for ?
```python
operator.attrgetter('license_text');

File: blackbelt/dependencies.py (hunk 2 above)
Comment (honzajavorek) [reply blackbelt/dependencies.py]: Yeah, right after dropping the Python 2.7 support 😅

Comment (char0n) [reply blackbelt/dependencies.py]: @honzajavorek dict comprehension is available in 2.* and also in 3.*

Comment (honzajavorek) [reply blackbelt/dependencies.py]: You are right 🙀 Fixed!

Comment (char0n) [reply blackbelt/dependencies.py]: It feels like dejavu with PHP. I remember telling to ppl, that I am a PHP developer. And if somebody new what PHP was, they usually asked: "What version?"  ;]

**Pull request merged**